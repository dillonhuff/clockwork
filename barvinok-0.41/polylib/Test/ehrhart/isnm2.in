
6 8
#  i  j  k  l   P  Q
0  3  0  2  1  -1  0   0	# 3i + 2k + l = P
0  4 -1  3  2   0 -1   0	# 4i - j + 3k + 2l = Q
1  1  0  0  0   0  0   0
1  0  1  0  0   0  0   0	# i,j,k,l >= 0
1  0  0  1  0   0  0   0
1  0  0  0  1   0  0   0

2 4
1  1  0   0		# P,Q >= 0
1  0  1   0

P Q
ISNM2 example
