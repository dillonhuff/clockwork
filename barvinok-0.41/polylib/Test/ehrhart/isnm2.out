---------------------------------------
Domain:
         4P -3Q  >= 0
         Q  >= 0
          1 >= 0

Vertices:
[ P/3,  4P/3-Q,  0,  0 ]
[ 0,  3P/2-Q,  P/2,  0 ]
[ 0,  2P-Q,  0,  P ]

Ehrhart Polynomial:
( 1/12 * P^2 + 1/2 * P + [ 1, 5/12, 2/3, 3/4, 2/3, 5/12 ]_P )

---------------------------------------
Domain:
         -4P + 3Q  >= 0
         3P -2Q  >= 0
          1 >= 0

Vertices:
[ 3P-2Q,  0,  -4P+3Q,  0 ]
[ P-Q/2,  0,  0,  -2P+3Q/2 ]
[ 0,  3P/2-Q,  P/2,  0 ]
[ 0,  2P-Q,  0,  P ]

Ehrhart Polynomial:
( -5/4 * P^2 + ( 2 * Q + 1/2 )
 * P + [ ( -3/4 * Q^2 + 0 * Q + [ 1, 3/4 ]_Q )
, ( -3/4 * Q^2 + 0 * Q + [ 3/4, 1/2 ]_Q )
 ]_P )

---------------------------------------
Domain:
         -3P + 2Q  >= 0
         2P - Q  >= 0
          1 >= 0

Vertices:
[ P-Q/2,  0,  0,  -2P+3Q/2 ]
[ 0,  2P-Q,  0,  P ]
[ 0,  0,  2P-Q,  -3P+2Q ]

Ehrhart Polynomial:
( 1 * P^2 + ( -1 * Q + 2 )
 * P + ( 1/4 * Q^2 + -1 * Q + [ 1, 3/4 ]_Q )
 )

